grafast/grafserv/src/utils.ts

```diff
@@ -228,10 +228,28 @@ export function makeGraphQLWSConfig(instance: GrafservBase): ServerOptions {
       const parseAndValidate = latestParseAndValidate;
 
       const parsedBody = parseGraphQLJSONBody(message.payload);
-      await hooks.process("processGraphQLRequestBody", {
-        body: parsedBody,
-        graphqlWsContext: ctx,
-      });
+      try {
+        await hooks.process("processGraphQLRequestBody", {
+          body: parsedBody,
+          graphqlWsContext: ctx,
+        });
+      } catch (e) {
+        if (e instanceof SafeError) {
+          return  [
+              new GraphQLError(
+                e.message,
+                null,
+                undefined,
+                undefined,
+                undefined,
+                e,
+                undefined,
+              ),
+            ]
+        } else {
+          throw e;
+        }
+      }
```

Comment:
Should this `try/catch` not also wrap the `validateGraphQLBody`, such that an invalid JSON object sent doesn't cause the connection to terminate too?

Under what circumstances would `processGraphQLRequestBody` throw where it wasn't a mistake from the client that should result in a connection reset?

Reply:
Yes I see, so, should we completely wrap the body of `onSubscribe` function as you initially suggested?
I think the connection should never be closed on throwing error, since the connection can be shared beetween several requests. (Maybe I wrong understand your question...)

Reply:
new commit